src/useResizeDetector.ts

```diff
@@ -42,7 +42,7 @@ function useResizeDetector<T extends HTMLElement = any>(
   useEnhancedEffect(() => {
     if (!handleWidth && !handleHeight) return;
 
-    const notifyResize = createNotifier(onResize, setSize, handleWidth, handleHeight);
+    const notifyResize = createNotifier(onResize, size, setSize, handleWidth, handleHeight);
```

Comment:
I just realized that I added `size` here without modifying the Effect dependencies, which we don't want to do, because that will re-create the ResizeObserver too frequently. Going to change my PR back to a Draft and take a look

Reply:
I've updated the PR and rebased it onto your most recent changes